src/rosdistro/distribution_cache.py

```diff
@@ -59,6 +59,7 @@ def __init__(self, name, data=None, distribution_file_data=None):
         self._distribution_file_data = data['distribution_file'] if data else distribution_file_data
         self.distribution_file = create_distribution_file(name, self._distribution_file_data)
         self.release_package_xmls = data['release_package_xmls'] if data else {}
+        self.source_repo_package_xmls = data['source_repo_package_xmls'] if data and 'source_repo_package_xmls' in data else {}
```

Comment:
Might this be a problem if the key exists but is empty? Then the variable will not be a dict as later expected.